xonsh/tools.py

```diff
@@ -517,37 +519,54 @@ def __iter__(self):
 RE_HIDDEN_MAX = re.compile('(\001.*?\002)+')
 
 
+_PT_COLORS={'BLACK': '#000000', 
+            'RED': '#FF0000',
+            'GREEN': '#008000',
+            'YELLOW':'#FFFF00', 
+            'BLUE'  :'#0000FF', 
+            'PURPLE':'#0000FF',
+            'CYAN':'#00FFFF',
+            'WHITE':'#FFFFFF'}
+
+_PT_STYLE = {'BOLD': 'bold',
+             'UNDERLINE':'underline',
+             'INTENSE':'italic'}
+
+
+def _make_style(color_name):
+    """ Convert color names to pygments styles codes """
+    style = []
+    for k,v in _PT_STYLE.items():
+        if k in color_name: 
+            style.append(v)
+    for k,v in _PT_COLORS.items():
+        if k in color_name:
+            style.append(v)
+    return ' '.join(style)
+    
+def get_xonsh_color_names(color_code):
+    """ Makes a reverse lookup in TERM_COLORS  """
+    try:
+        return next(k for k, v in TERM_COLORS.items() if v == color_code)
+    except StopIteration:
+        return 'No_Color'
+
+
+     
 def format_prompt_for_prompt_toolkit(prompt):
-    """Uses workaround for passing a string with color sequences.
-
-    Returns list of characters of the prompt, where some characters can be not
-    normal characters but FakeChars - objects that consists of one printable
-    character and escape sequences surrounding it.
-    Returned list can be later passed as a prompt to prompt_toolkit.
-    If prompt contains no printable characters returns equivalent of empty
-    string.
+    """Converts a prompt with color codes to a pygments style and tokens
     """
-    def append_escape_seq(lst, suffix):
-        last = lst.pop()
-        if isinstance(last, FakeChar):
-            lst.append(FakeChar(last.char, prefix=last.prefix, suffix=suffix))
-        else:
-            lst.append(FakeChar(last, suffix=suffix))
-    pos = 0
-    match = RE_HIDDEN_MAX.search(prompt, pos)
-    if match and match.group(0) == prompt:
-        return ['']
-    formatted_prompt = []
-    while match:
-        formatted_prompt.extend(list(prompt[pos:match.start()]))
-        pos = match.end()
-        if not formatted_prompt:
-            formatted_prompt.append(FakeChar(prompt[pos],
-                                             prefix=match.group(0)))
-            pos += 1
-        else:
-            append_escape_seq(formatted_prompt, match.group(0))
-        match = RE_HIDDEN_MAX.search(prompt, pos)
+    parts = RE_HIDDEN_MAX.split(prompt)
+    #ensure that parts is [colorcode, string, colorcode, string,...]
+    if parts and parts[0] is '': 
```

Comment:
I see why  `parts[0] is ''` is not a good idea. But isn't `if parts` the pythonic way to test if a list is empty.